gallery/tutorials/tutorials/micrograph_source.py

```diff
@@ -0,0 +1,220 @@
+
+# Create a micrograph with a negative boundary, allowing particles to generate outward.
+outbound_micrograph = MicrographSource(
+    colliding_sim,
+    boundary=-20,
+    interparticle_distance=1,
+    noise_adder=noise,
+    particles_per_micrograph=n_particles_per_micrograph,
+    micrograph_size=500,
+    micrograph_count=n_micrographs,
+)
+
+# Plot the micrographs
+outbound_micrograph.micrographs[:].show()
+
+# %%
+# Particle IDs
+# ------------
+# Each particle comes from the simulation, and has its own ID relative to its ``Simulation`` and micrograph.
+# Let's choose 4 random numbers, as our global (Simulation) particle IDs from ``test_micrograph=1``.
+test_micrograph = 1
+local_particle_ids = np.random.choice(n_particles_per_micrograph, 4)
+print(f"Local particle IDS: {local_particle_ids}")
+
+# %%
+# We can obtain the images from our MicrographSource by retrieving their centers and plotting the boundary boxes.
+centers = np.zeros((len(local_particle_ids), 2), dtype=int)
+for i in range(local_particle_ids.shape[0]):
+    centers[i] = colliding_micrograph.centers[test_micrograph][local_particle_ids[i]]
+
+# Let's use the particles' centers and sizes to perform "perfect particle picking" on this test micrograph.
+p_size = colliding_micrograph.particle_box_size
+micrograph_picked_particles = np.zeros(
+    (
+        len(local_particle_ids),
+        colliding_micrograph.particle_box_size,
+        colliding_micrograph.particle_box_size,
+    )
+)
+
+for i, center in enumerate(centers):
+    x, y = center[0], center[1]
+    # Calculate the square of the particle
+    particle = colliding_micrograph.clean_micrographs[test_micrograph].asnumpy()[0][
+        x - p_size // 2 : x + p_size // 2, y - p_size // 2 : y + p_size // 2
+    ]
+    micrograph_picked_particles[i] = particle
+
+# Let's plot and look at the particles!
+Image(micrograph_picked_particles)[:].show()
+
+# %%
+# The simulated particles are inverted from the MRC due to convention, as we're trying to measure interference.
+# Let's find the images from the ``Simulation`` using the ``get_particle`` method to retrieve their global IDs.
+global_particle_ids = np.zeros((len(local_particle_ids)), dtype=int)
+for i in range(len(local_particle_ids)):
+    global_particle_ids[i] = colliding_micrograph.get_particle(
+        test_micrograph, local_particle_ids[i]
+    )
+
+# Plot the simulation's images
+colliding_sim.images[global_particle_ids].show()
+
+# %%
+# We can check if these global IDs match our local particle IDs with the ``get_micrograph`` method.
+check_local_ids = np.zeros((len(local_particle_ids)), dtype=int)
+for i in range(len(local_particle_ids)):
+    # Get each particle's corresponding micrograph ID and local particle ID
+    micrograph_id, check_local_ids[i] = colliding_micrograph.get_micrograph(
+        global_particle_ids[i]
+    )
+    assert micrograph_id == 1
+np.testing.assert_array_equal(local_particle_ids, check_local_ids)
```

Comment:
Maybe `np.allclose`?

Reply:
These are integers, so I think I'll leave it for now.